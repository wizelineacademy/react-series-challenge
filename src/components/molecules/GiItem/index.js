import React, { Component } from "react";
import styled from "styled-components";

const StyledImg = styled.div`
  img {
    border: 1em solid red;
    border-radius: 1em;
  }
`;

class GiItem extends Component {
  constructor(props) {
    super(props);
    this.getClicked = this.getClicked.bind(this);
  }

  getClicked() {
    this.props.toggleFavs(this.props.giphy);
  }

  render() {
    const { giphy, selectedFavs } = this.props;

    const img = giphy.images["fixed_height"];

    const isFav =
      selectedFavs !== undefined
        ? selectedFavs.indexOf(giphy.id) !== -1
        : false;

    if (isFav) {
      return (
        <StyledImg>
          <img src={img.url} alt={giphy.title} onClick={this.getClicked} />
          <p>Selected</p>
        </StyledImg>
      );
    }

    return <img src={img.url} alt={giphy.title} onClick={this.getClicked} />;
  }
}

export default GiItem;
