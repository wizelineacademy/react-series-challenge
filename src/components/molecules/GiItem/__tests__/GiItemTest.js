import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import GiItem from "../index";
import giphyMock from "./mocks/data.json";

describe("GiItem", () => {
  it("should render without crashing", () => {
    const testGiItem = shallow(<GiItem giphy={giphyMock.data} />);

    expect(shallowToJson(testGiItem)).toMatchSnapshot();
  });
});
