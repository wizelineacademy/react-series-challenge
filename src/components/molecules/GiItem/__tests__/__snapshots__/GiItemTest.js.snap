// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiItem should render without crashing 1`] = `
<img
  alt="palm trees storm GIF"
  onClick={[Function]}
  src="https://media3.giphy.com/media/HmTLatwLWpTQk/200.gif"
/>
`;
